syntax = "proto3";

package confuse.client.v1;

// ClientConnector service - MCP client connection management
service ClientConnector {
  // Session management (internal use)
  rpc CreateSession(CreateSessionRequest) returns (Session);
  rpc GetSession(GetSessionRequest) returns (Session);
  rpc CloseSession(CloseSessionRequest) returns (CloseSessionResponse);
  
  // Session status
  rpc ListActiveSessions(ListSessionsRequest) returns (ListSessionsResponse);
}

// Messages
message CreateSessionRequest {
  string client_id = 1;
  string client_type = 2;  // claude, cursor, windsurf, etc.
  map<string, string> metadata = 3;
}

message Session {
  string session_id = 1;
  string client_id = 2;
  string client_type = 3;
  string status = 4;
  int64 created_at = 5;
  int64 last_activity = 6;
}

message GetSessionRequest {
  string session_id = 1;
}

message CloseSessionRequest {
  string session_id = 1;
}

message CloseSessionResponse {
  bool success = 1;
}

message ListSessionsRequest {
  optional string client_type = 1;
}

message ListSessionsResponse {
  repeated Session sessions = 1;
  int32 total = 2;
}
